File: Easy/Filter/String/1941. Check if All Characters Have Equal Number of Occurrences.py
"""
https://leetcode.com/problems/check-if-all-characters-have-equal-number-of-occurrences/
"""


def areOccurrencesEqual(s: str) -> bool:
    count = {}
    for i in s:
        count[i] = count.get(i, 0) + 1
    return len(set(count.values())) == 1


# s = "abacbc"
s = "aaabb"
print(areOccurrencesEqual(s))

File: PHP/Easy/Filters/Hash_Table/1941_Check_if_All_Characters_Have_Equal_Number_of_Occurrences.php
<?php
// https://leetcode.com/problems/check-if-all-characters-have-equal-number-of-occurrences/
function areOccurrencesEqual($s)
{
    $count = [];
    foreach (str_split($s) as $c) {
        if (array_key_exists($c, $count)) {
            $count[$c] += 1;
        } else {
            $count[$c] = 1;
        }
    }
    $fre = array_values($count);
    foreach ($fre as $i) {
        if ($i !== $fre[0]) {
            return false;
        }
    }
    return true;

}


$s = "aabb";
print_r(areOccurrencesEqual($s));